core/src/main/java/com/opensymphony/xwork2/util/AbstractLocalizedTextProvider.java

```diff
@@ -413,11 +411,11 @@ public ResourceBundle findResourceBundle(String aBundleName, Locale locale) {
                     }
                 } catch (MissingResourceException e) {
                     LOG.debug("Missing resource bundle [{}]!", aBundleName, e);
-                    missingBundles.add(key);
+                    missingBundles.put(key, Boolean.TRUE);
```

Comment:
Consider changing `put()` to `putIfAbsent()` as the closest behaviour match to the previous `add()` method usage:
`missingBundles.putIfAbsent(key, Boolean.TRUE);`

Reply:
Not sure that it makes any difference, but let's stick to the closest behaviour as mentioned.